### ISSUE TITLE
RFE: autobooting with a workbench for the non-bootable hdf images?

### ISSUE THREAD

**Issue by i30817** [libretro/libretro-uae#243]
It's a bit annoying that there are still hd images around that are not WHDLoad ready, so a option to fallback to loading a workbench if the hd is not bootable and there is no whdload boot file present to manually navigate the workbench to the game and boot it up would be cool.

Not sure if a good idea, since ofc, there are different WB for different models and how savestates would react.

Or we can wait until people stop creating lha/hdf that are not bootable and don't have the whdload file; though the problem appears to be that the people who made these (often hacks/remakes, like the Heart of the Alien Redux) do it and then no-one ever dares to distribute a modified version, even for convenience.

**Comment by WindyWinston** [comment]
I think they do it because Workbench and Kickstarts are still illegal to distribute. Cloanto have the licence to these files and are still making money from Amiga Forever.

**Comment by i30817** [comment]
I understand that, but lha doesn't actually need the workbench to start, just the kickstart, so i'd expect turning a 'hdloader' into whdload would be trivial by adding the launcher file... but maybe i'm wrong, since i never did it.

**Comment by sonninnos** [comment]
Might be easiest to simply edit those lazy images to have the startup-sequence, since that will already be searched and launched. But I guess it isn't impossible to start WB when slaves and startup-sequences are not found. I'll have a look soonish.

**Comment by i30817** [comment]
BTW, there is a particular whdload that might be handy to special case:

http://www.whdload.de/apps/Workbench31.html

Yes, this is probably a bit silly but might be a good way to standardize. Or not and just having the floppies would be enough...

**Comment by sonninnos** [comment]
How about this tiny reorganizing with the same tools that already were there:
[WHDLoad.zip](https://github.com/libretro/libretro-uae/files/4192567/WHDLoad.zip)

No need to go full WB when the simple selector should be enough, amirite.

**Comment by i30817** [comment]
Whatever works. Just to be sure, this means that it's still going to use the core inbuilt selector even if it doesn't find a .slave file by allowing selecting executable files?

**Comment by sonninnos** [comment]
The selector initially shows only the info/icon files, but navigating with it will show all files.

Just replace the existing one with that for testing. If you find it adequate I will of course update it into the core.

**Comment by i30817** [comment]
I figured, i'm compiling RA atm and it's not building from upstream so it might take a while to get back to you. But if you tested it works i'm already satisfied.

**Comment by i30817** [comment]
Ok i tested it and it works well except it could probably use a tweak if possible.

Selecting a 'Readme.info' file and quitting it starts it up again, so you have to restart to select the game .info that starts the actual game.

There was also a curious bug on HeartOfTheAlienRedux_v1.1_HD_DaliborJavorcek.lha that i'm not sure is original, where selecting the HeartOfTheAlienRedux.info started the hack-intro but quitting failed at starting the game executable (it shows a 'cannot open mathiassomething library' and a 'error 900' at the next line) and starts up the readme.info instead (and then you're at the first bug). Trying to startup the executable that the game.info would start is also weird because it's not visible on the root at first, you have to enter a dir and get back up to show all files that aren't a info or a dir (on purpose maybe?). Also it has two game.info files which might influence (you can only tell after you disable that 'only show .info files and dirs) with a bit of directory movement).

In short that game may be broken on the environment / it expects something from workbench. Not a big deal, just funny and probably maybe showing something to fix with the 'retry other info files' strange behavior going on).

**Comment by sonninnos** [comment]
Added the missing library, fixed the scripting issues and got HeartOfTheAlienRedux working by forcing it to A1200.

I'll soon PR it, but here is the test material anyway: 
[WHDLoad.zip](https://github.com/libretro/libretro-uae/files/4194783/WHDLoad.zip)

**Comment by i30817** [comment]
Hopefully retroarch/libretro upstream gets per playlist config overrides someday so i can force A1200 for the retroplay 'HD Loaders' playlist and leave the default A500 for the retroplay 'whdload games' playlist.
